### ISSUE TITLE
Add token creation support

### ISSUE THREAD

**Issue by ConnorMac** [rehive/service-stellar#8]
The service needs full support for token/asset creation by company admins:

To do this we will:

**Create a distribution account on activation**
 -

- [ ] Generate keypair, store in new DistributionAccount model
- [ ] Create Rehive account and store reference

**Create trust between issuer and distribution account**
- 

Endpoints:
**/admin/asset/<id>/**
[GET]
Issuer_address
Currency_code
Balance on Rehive distribution account

**/admin/asset/** 
[POST]:
issuer_address (Public key of issuer)
currency_code (3-12 Alphanumeruc)

An admin will post to the above endpoint, we will then:

- [ ] Broadcast a trust transaction from the distribution account
- [ ] Create the currency on the distribution account created on Rehive

**Create endpoint for an admin to fund the tokens**
-
The main concern here is the admin will need to post their issuer private key. We MUST NOT store this as this is already a security risk.

Endpoints:
**/admin/asset/<id>/fund/**
 [POST]:
Issuer private key
Amount

- [ ] Broadcast a funding transaction for the asset using the private issuer key for the amount specified
- [ ] Update our transaction monitoring to look for other assets and update the distribution account balance on Rehive when nescessary

**Resources**
-

Stellar blog post of on tokens: [https://www.stellar.org/blog/tokens-on-stellar/](https://www.stellar.org/blog/tokens-on-stellar/)
Gist using stellar python library: https://gist.github.com/ConnorMac/ea1491ffa2015883e1140ca5f85d5a04

**Comment by michailbrynard** [comment]
@ConnorMac Thanks for the writeup. Regarding DistributionAccount model: could we not rather just have a hot wallet model (as we do in the bitcoin service) and use this for distribution, sending and everything?

**Comment by ConnorMac** [comment]
@michailbrynard I agree, I think for the moment it is probably worth running all the distribution of assets through a single account and it could help us move toward some of the hotwallet structure on this service as well.

**Comment by ConnorMac** [comment]
An interesting case to consider: https://stellar.stackexchange.com/questions/254/can-i-create-a-non-fungible-token/257 . It would be nice if we can specify non-fungible asset flag when creating an asset and the handle/force lowest denomination transactions.
